### PR TITLE
[86b2dy6v0] Contact Page Backend

### PR DESCRIPTION
**Pull request by fizo-neechan**
- Moved contactSchema to types, made a type aswell using z.typeof
- Added the mongoose models and the dbConnect functions
- Added middleware for /api routes to always check for dbConnect
- made the api and controller for `/api/contact` route

### CODE REVIEW

**Comment by anasbinarif** [comment]
@fizo-neechan can you kindly see why you have Mehnaz commits in this PR?

File: .eslintrc.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   },
   "plugins": ["react", "import", "@next/next", "@typescript-eslint"],
   "rules": {
-    "no-console": ["error"],
+    "no-console": ["warn"],
```
Comment (anasbinarif) [inline .eslintrc.json:26]: kindy set it back to error as we don't want console logs in deployment

**Comment by fizo-neechan** [comment]
> @fizo-neechan can you kindly see why you have Mehnaz commits in this PR?

I've updated you on slack regarding this since the Backend had a dependency on mehnaz's work, so I had to branch from her branch since this feature was not yet complete. As soon as her branch get's merged, I am sure the commits from her side will dissapear

File: src/app/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+import type { NextRequest } from 'next/server';
+
+import dbConnect from '@/utils/mongodb';
+
+export async function middleware(request: NextRequest) {
+  if (request.nextUrl.pathname.startsWith('/api')) {
+    try {
+      await dbConnect();
+    } catch (error) {
+      console.error('Database connection error:', error);
```
Comment (anasbinarif) [inline src/app/middleware.ts]: remove console errors

File: src/features/auth/login/LoginModal.tsx [hunk 3]
```diff
@@ -49,14 +50,14 @@ const LoginModal: React.FC<LoginModalProps> = ({ open, handleClose, onSwitchToSi
     setLoading(true);
     try {
       await new Promise((resolve) => setTimeout(resolve, 2000));
-      setSnackbarMessage('Login successful');
+      showSnackbar('Login successful');
       reset();
       // eslint-disable-next-line no-console
       console.log('Form data:', data);
```
Comment (anasbinarif) [inline src/features/auth/login/LoginModal.tsx]: here too

File: .eslintrc.json (hunk 1 above)
Comment (fizo-neechan) [reply .eslintrc.json:26]: resolved in [f5e7758](https://github.com/anasbinarif/ace-my-exam/pull/10/commits/f5e7758c9a2c68e71e1d45f4114d9629a3a87510)

File: src/app/middleware.ts (hunk 2 above)
Comment (fizo-neechan) [reply src/app/middleware.ts]: resolved in [f5e7758](https://github.com/anasbinarif/ace-my-exam/pull/10/commits/f5e7758c9a2c68e71e1d45f4114d9629a3a87510)

File: src/features/auth/login/LoginModal.tsx (hunk 3 above)
Comment (fizo-neechan) [reply src/features/auth/login/LoginModal.tsx]: resolved in [f5e7758](https://github.com/anasbinarif/ace-my-exam/pull/10/commits/f5e7758c9a2c68e71e1d45f4114d9629a3a87510)

**Comment by anasbinarif** [comment]
@fizo-neechan let me know when u integrate SQL, rest is all good

File: src/app/api/contact/route.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  await initializeDataSource();
+  try {
+    const body = await request.json();
+    const validatedData = contactSchema.parse(body);
+
+    const contact = await createContact(validatedData);
+
+    return NextResponse.json(contact, { status: 201 });
+  } catch (error: unknown) {
+    // console.error(error);
```
Comment (anasbinarif) [inline src/app/api/contact/route.ts]: can u please remove these comments?

File: src/features/contact/ContactForm.hooks.ts
Comment (anasbinarif) [inline src/features/contact/ContactForm.hooks.ts:1]: I would suggest moving all the settings and calling the endpoints to a separate folder, you can name it accordingly, so in the component, we will just call the react query function.
will be beneficial for letter usage 
thanks

**Review by anasbinarif — CHANGES_REQUESTED** [review]

File: src/app/api/contact/route.ts (hunk 4 above)
Comment (fizo-neechan) [reply src/app/api/contact/route.ts]: done

File: src/features/contact/ContactForm.hooks.ts
Comment (fizo-neechan) [reply src/features/contact/ContactForm.hooks.ts:1]: Good point! I've done the changes in my latest commit, moving the hooks to `@/hooks`

**Review by anasbinarif — APPROVED** [review]

**Pull request merged**